### PR TITLE
cows, cows everywhere

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
Hm. Looks like "Inflect::to_snake_case" =/= "convert_case::Case::Snake"
```rust
    let string = "_1P8_READY";
    assert_eq!(string.to_snake_case(), string.to_case(Case::ScreamingSnake));
```

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"_1p8_ready"`,
 right: `"1_P_8_READY"`', src/main.rs:6:5
```

File: ci/svd2rust-regress/src/main.rs [hunk 1]
```diff
@@ -13,92 +14,91 @@ use std::path::PathBuf;
...
 
     /// Print all available test using the specified filters
-    #[structopt(long = "list")]
+    #[clap(long)]
     list: bool,
 
     /// Path to an `rustfmt` binary, relative or absolute.
     /// Defaults to `$(rustup which rustfmt)`
-    #[structopt(long = "rustfmt_bin_path", parse(from_os_str))]
+    #[clap(long = "rustfmt_bin_path")]
```
Comment (Emilgardis) [inline ci/svd2rust-regress/src/main.rs:71]:
```suggestion
    #[clap(long)]
```

`svd2rust` used, with `structopt`, snake case

File: ci/svd2rust-regress/src/main.rs [hunk 2]
```diff
@@ -13,92 +14,91 @@ use std::path::PathBuf;
...
+        short = 'a',
         long = "architecture",
-        raw(validator = "validate_architecture")
+        value_parser = validate_architecture,
     )]
     arch: Option<String>,
 
     /// Include tests expected to fail (will cause a non-zero return code)
-    #[structopt(short = "b", long = "bad-tests")]
+    #[clap(short = 'b', long = "bad-tests")]
```
Comment (Emilgardis) [inline ci/svd2rust-regress/src/main.rs:58]:
```suggestion
    #[clap(short = 'b')]
```

File: ci/svd2rust-regress/src/main.rs [hunk 3]
```diff
@@ -13,92 +14,91 @@ use std::path::PathBuf;
...
 use std::time::Instant;
-use structopt::StructOpt;
 
-#[derive(StructOpt, Debug)]
+#[derive(Parser, Debug)]
 #[structopt(name = "svd2rust-regress")]
 struct Opt {
     /// Run a long test (it's very long)
-    #[structopt(short = "l", long = "long-test")]
+    #[clap(short = 'l', long = "long-test")]
```
Comment (Emilgardis) [inline ci/svd2rust-regress/src/main.rs:22]:
```suggestion
    #[clap(short = 'l')]
```

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: ci/svd2rust-regress/src/main.rs (hunk 1 above)
Comment (burrbull) [reply ci/svd2rust-regress/src/main.rs:71]: ?

Comment (Emilgardis) [reply ci/svd2rust-regress/src/main.rs:71]: the flag should be `--rustfmt-bin-path` not `--rustfmt_bin_path`

Comment (burrbull) [reply ci/svd2rust-regress/src/main.rs]: done, rebased

**Review by Emilgardis — APPROVED** [review]
bors r+
